mcstatus/server.py

```diff
@@ -13,6 +15,17 @@
 __all__ = ["MinecraftServer", "MinecraftBedrockServer"]
 
 
+def ensure_valid_ip(host: object, port: object):
+    if not isinstance(host, str):
+        raise TypeError(f"Host must be a string address, got {type(host)} ({host!r})")
+    if not isinstance(port, int):
+        raise TypeError(f"Port must be an integer port number, got {type(port)} ({port})")
+    if port >= 36635 or port < 0:
+        raise ValueError(f"Port must be within the allowed range (0-2^16), got {port}")
+    if not re.fullmatch(r"(\w+\.)*\w+", host):
```

Comment:
We could take from this public post that's kindly provided an RFC compliant regex
```suggestion
    if not re.fullmatch(r"(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])", host):
```
https://stackoverflow.com/a/106223/2607949

Or we could do an immediate DNS resolution to see if a host resolves. Possibly initializing `self.ip` (or something like that) in the process. Validating that a provided host string is compliant with an RFC does not imply that the DN exists on DNS servers. However, doing the resolution here instead of in the individual methods that are calling `dns.resolver.resolve` could harm keeping a long-running instance of MinecraftServer. DN are allowed to change over time, so doing the resolution queries every time ping or query is invoked is a good thing.

:+1: on the regex here instead of DNS resolution.